core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -517,40 +524,48 @@ private HttpResponse newResponse(HttpMethod method, boolean isHealthy) {
         }
     }
 
-    private AggregatedHttpResponse getResponse(boolean isHealthy) {
-        if (isHealthy) {
+    private AggregatedHttpResponse getResponse(HealthStatus healthStatus) {
```

Comment:
What if we add additional `SERVER_STOPPING` status. That way, we can make `getResponse` method much clear.

Reply:
I kind of though `UNDER_MAINTENANCE` is a generalized version of that. What do you think about adding it once we really have a use case for it?

Reply:
Sure, let's add it! 😄